src/librustdoc/html/render/write_shared.rs

```diff
@@ -125,21 +125,19 @@ impl Context<'_> {
         minify: bool,
         emit: &[EmitType],
     ) -> Result<(), Error> {
-        let tmp;
-        let contents = if minify {
-            tmp = if resource.extension() == Some(&OsStr::new("css")) {
+        if minify {
+            let contents = if resource.extension() == Some(&OsStr::new("css")) {
                 minifier::css::minify(contents).map_err(|e| {
                     Error::new(format!("failed to minify CSS file: {}", e), resource.path(self))
                 })?
             } else {
                 minifier::js::minify(contents)
             };
-            tmp.as_bytes()
+            self.write_shared(resource, contents, emit)
         } else {
-            contents.as_bytes()
-        };
-
-        self.write_shared(resource, contents, emit)
+            // FIXME: this copy might be avoidable?
+            self.write_shared(resource, contents.to_string(), emit)
```

Comment:
What error do you get if you don't use `.to_string()`? `contents` is moved inside this function after all so it seems a bit strange, no?

Reply:
`write_shared` takes `impl 'static + Send + AsRef<[u8]>`, so that rayon can spawn a new thread for it. `contents` could be a reference to a temporary.
```
error[E0621]: explicit lifetime required in the type of `contents`
   --> src/librustdoc/html/render/write_shared.rs:139:13
    |
139 |             self.write_shared(resource, contents, emit)
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'static` required
```

Reply:
fixed it in the latest commit :)